src/dcqc/reports.py

```diff
@@ -46,6 +50,12 @@ def to_file(self, obj: Any, url: str, overwrite: bool):
         #       (e.g., relativize them based on output JSON path)
         with fs.open(fs_path, "w") as outfile:
             json.dump(obj, outfile, indent=2)
+            outfile.write("\n")
+
+    def _generate_single(self, item: SerializableMixin) -> SerializedObject:
+        item.serialize_paths_relative_to(self.paths_relative_to)
+        report = item.to_dict()
```

Comment:
Would it make sense to include the `serialize_paths_relative_to` in the `to_dict` function?

Reply:
I considered that, but I didn't want to change the method signature in all places where `to_dict()` is defined. As I mentioned in a `TODO`, I think this logic ultimately belongs in the `JsonReport` class (probably going to rename it to `JsonExporter`) as a custom `JSONEncoder` subclass. I think Paths should be converted to strings as late as possible (_i.e._ during JSON serialization and not in `to_dict()`). This was a quick fix, but I'm not super happy with it. I will likely refactor as soon as I touch the `JsonReport` code.

Reply:
👍🏼